gobblin-data-management/src/main/java/gobblin/data/management/copy/replication/CopySourceWithMultipleSourceLocation.java

```diff
@@ -0,0 +1,277 @@
+import gobblin.config.client.ConfigClient;
+import gobblin.config.client.api.ConfigStoreFactoryDoesNotExistsException;
+import gobblin.config.client.api.VersionStabilityPolicy;
+import gobblin.config.store.api.ConfigStoreCreationException;
+import gobblin.config.store.api.VersionDoesNotExistException;
+import gobblin.configuration.ConfigurationKeys;
+import gobblin.configuration.SourceState;
+import gobblin.configuration.State;
+import gobblin.configuration.WorkUnitState;
+import gobblin.data.management.copy.CopyableDatasetBase;
+import gobblin.data.management.copy.extractor.EmptyExtractor;
+import gobblin.data.management.copy.extractor.FileAwareInputStreamExtractor;
+import gobblin.data.management.copy.publisher.CopyEventSubmitterHelper;
+import gobblin.data.management.copy.watermark.CopyableFileWatermarkGenerator;
+import gobblin.data.management.copy.watermark.CopyableFileWatermarkHelper;
+import gobblin.data.management.dataset.DatasetUtils;
+import gobblin.data.management.partition.FileSet;
+import gobblin.data.management.retention.profile.ConfigurableGlobDatasetFinder;
+import gobblin.dataset.Dataset;
+import gobblin.dataset.DatasetsFinder;
+import gobblin.dataset.IterableDatasetFinder;
+import gobblin.dataset.IterableDatasetFinderImpl;
+import gobblin.instrumented.Instrumented;
+import gobblin.metrics.GobblinMetrics;
+import gobblin.metrics.MetricContext;
+import gobblin.metrics.Tag;
+import gobblin.metrics.event.EventSubmitter;
+import gobblin.metrics.event.sla.SlaEventKeys;
+import gobblin.source.extractor.Extractor;
+import gobblin.source.extractor.WatermarkInterval;
+import gobblin.source.extractor.extract.AbstractSource;
+import gobblin.source.workunit.Extract;
+import gobblin.source.workunit.WorkUnit;
+import gobblin.source.workunit.WorkUnitWeighter;
+import gobblin.util.ExecutorsUtils;
+import gobblin.util.HadoopUtils;
+import gobblin.util.PathUtils;
+import gobblin.util.WriterUtils;
+import gobblin.util.binpacking.FieldWeighter;
+import gobblin.util.binpacking.WorstFitDecreasingBinPacking;
+import gobblin.util.executors.IteratorExecutor;
+import gobblin.util.guid.Guid;
+import gobblin.util.iterators.InterruptibleIterator;
+import gobblin.data.management.copy.ConcurrentBoundedWorkUnitList;
+import gobblin.data.management.copy.CopyConfiguration;
+import gobblin.data.management.copy.CopySource;
+
+@Slf4j
+public class CopySourceWithMultipleSourceLocation extends CopySource{
+
+  public static final String CONFIG_STORE_ROOT = "config.store.root";
+  public static final String CONFIG_STORE_REPLICATION_ROOT = "config.store.replication.root";
+  public static final String CONFIG_STORE_REPLICATION_TAG = "config.store.replication.tag";
+
+  public static void displayAllConfig(Config c){
+    for(Map.Entry<String, ConfigValue> entry :c.entrySet()){
+      log.info("AAA key " + entry.getKey() + " value " + entry.getValue());
+    }
+  }
+
+  private DataFlowTopology.CopyRoute getCopyRoute(DataFlowTopology topology)throws IOException{
```

Comment:
What is this method supposed to do? Does it determine which route to use for a dataset? If yes, it seems to make 2 assumptions:
- We are always running in pull mode (it uses the default cluster)
- There is only one ReplicationEntryPoint per cluster. 

Both of those assumptions are incorrect.